compose/service/record.go

```diff
@@ -435,6 +442,43 @@ func (svc record) findSensitive(ctx context.Context, userID uint64, namespace *t
 	return
 }
 
+// SearchRevisions returns iterator for revisions of a record
+//
+func (svc record) SearchRevisions(ctx context.Context, namespaceID, moduleID, recordID uint64) (dal.Iterator, error) {
+	var (
+		aProps = &recordActionProps{record: &types.Record{NamespaceID: namespaceID, ModuleID: moduleID, ID: recordID}}
+
+		rec  *types.Record
+		iter dal.Iterator
+		mod  *types.Module
+		ns   *types.Namespace
+	)
+
+	err := func() (err error) {
+		ns, mod, rec, err = loadRecordCombo(ctx, svc.store, svc.dal, namespaceID, moduleID, recordID)
+		if err != nil {
+			return
+		}
+
+		aProps.setModule(mod)
+		aProps.setNamespace(ns)
+		aProps.setRecord(rec)
+
+		if !mod.ModelConfig.Revisions.Enabled {
+			return RecordErrRevisionsDisabledOnModule()
+		}
+
+		if !svc.ac.CanSearchRevisionsOnRecord(ctx, rec) {
+			return RecordErrNotAllowedToSearchRevisions()
+		}
```

Comment:
should we also have a `read` perm. for specific revisions?
I'd say no, but to consider

Reply:
no